docs/annotations.md

```diff
@@ -4,16 +4,129 @@
 and [annotations :octicons-link-external-16:](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
 are used to attach additional metadata information to Kubernetes resources.
 
-Labels and annotations are rather similar. The difference between them is that
-labels are used by Kubernetes to identify and select objects, while annotations
-are assigning additional *non-identifying* information to resources.
-Therefore, typical role of Annotations is facilitating integration with some
-external tools.
+Labels and annotations are rather similar but differ in purpose.
+
+**Labels** are used by Kubernetes to identify and select objects. They enable filtering and grouping, allowing users to apply selectors for operations like deployments or scaling.
+
+**Annotations** are assigning additional *non-identifying* information that doesn't affect how Kubernetes processes resources. They store descriptive information like deployment history, monitoring configurations or external integrations.
+
+The following diagram illustrates this difference:
+
+
+```mermaid
+%%{ init : { "theme": "base", "themeVariables": {
+  "primaryColor": "#93D0FF",
+  "primaryTextColor": "#ffffff",
+  "lineColor": "#ffffff",
+  "background": "#1e1e1e"
+} }}%%
+graph TD
+    A[Custom Resource] --> B[Operator]
+    B --> C[Kubernetes resources]
+    C --> D[Labels]
+    C --> E[Annotations]
+    D --> F[Selection]
+    D --> G[Grouping]
+    E --> H[External tools]
+    E --> I[Documentation]
+```
+
+Both Labels and Annotations are assigned to the following objects managed by Percona Operator for PostgreSQL:
+
+* Custom Resource Definitions
+* Custom Resources
+* Deployments
+* Services
+* StatefulSets
+* PVCs
+* Pods
+* ConfigMaps and Secrets
+
+## When to use labels and annotations
+
+Use **Labels** when:
+
+* The information is used for object selection
+* The data is used for grouping or filtering
+* The information is used by Kubernetes controllers
+* The data is used for operational purposes
+
+Use **Annotations** when:
+
+* The information is for external tools
+* The information is used for debugging
+* The data is used for monitoring configuration
+
+## Labels and annotations used by Percona Operator for MySQL
```

Comment:
Here we reference MySQL Operator, but it should be Postgres

Reply:
Thanks, fixed